### PR TITLE
Fix SQL removing related row from many side 

### PR DESCRIPTION
**Pull request by mike12345567**
## Description
This fixes an issue raised by @ConorWebb96 when attempting to remove related rows from a relationship specifically from the many side. Adding a test case to cover this in future.

## Addresses
- https://linear.app/budibase/issue/BUDI-8715/there-seems-to-be-issues-with-relationships-in-budibase-at-the-moment
- https://github.com/Budibase/budibase/issues/14747

### CODE REVIEW

File: packages/server/src/api/controllers/row/ExternalRequest.ts [hunk 1]
```diff
@@ -743,9 +745,19 @@ export class ExternalRequest<T extends Operation> {
...
-      : await aliasing.queryWithAliasing(json, makeExternalQuery)
+    let response: DatasourcePlusQueryResponse
+    // there's a chance after input processing nothing needs updated, so pass over the call
+    // we might still need to perform other operations like updating the foreign keys on other rows
+    if (
+      this.operation === Operation.UPDATE &&
+      Object.keys(row || {}).length === 0
+    ) {
+      response = [config.row!]
+    } else {
```
Comment (samwho) [inline packages/server/src/api/controllers/row/ExternalRequest.ts:756]: I don't think I fully understand this 🤔 So if the row is empty we return `[{}]`?

File: packages/backend-core/src/sql/sql.ts [hunk 2]
```diff
@@ -325,14 +325,18 @@ class InternalBuilder {
...
+        const { column } = this.splitter.run(key)
+        const schema = this.table.schema[column]
+        if (!schema) {
+          continue
+        }
+        body[key] = this.parse(value, schema)
       }
-      body[key] = this.parse(value, schema)
+    } catch (err) {
+      console.log(err)
```
Comment (samwho) [inline packages/backend-core/src/sql/sql.ts:339]: Is it correct to just return the body as-is, un-parsed, when anything goes wrong in parsing? Feels like something that's destined to cause future bugs.

File: packages/server/src/api/controllers/row/ExternalRequest.ts (hunk 1 above)
Comment (mike12345567) [reply packages/server/src/api/controllers/row/ExternalRequest.ts:756]: Ah - in this case the `row` is the processed row, which may have been stripped down if nothing has changed, while `config.row` is unprocessed - I'll rename to make it more obvious.

File: packages/backend-core/src/sql/sql.ts (hunk 2 above)
Comment (mike12345567) [reply packages/backend-core/src/sql/sql.ts:339]: Oh woops - I should have removed this try-catch, was just during my debugging, nice catch!

**Review by samwho — APPROVED** [review]

**Pull request merged**